pkg/terraform/parser/parser.go

```diff
@@ -115,6 +121,26 @@ func (p *TerraformParser) StateFileURIs(ctx context.Context, gcsBuckets []string
 	return gcsURIs, nil
 }
 
+// StateWithoutResources determines if the given statefile at the uri contains any resources or not.
+func (p *TerraformParser) StateWithoutResources(ctx context.Context, uri string) (bool, error) {
```

Comment:
Why isnt this functionality part of the process states logic? Seems like a wasted download call here?

Reply:
Because ProcessStates is actually only concerned with IAM state. That could change if we make drift detection do more than just IAM. We don't use ProcessStates for statefile drift detection. Currently, we only use GCS to check which files actually exist. This adds functionality to check if the statefile has NO state.

Perhaps we should rename ProcessStates to ProcessIAMStates or something? I can do it in a follow up